src/Tools/CoreFx.Tools/GenerateSlnFile.cs

```diff
@@ -0,0 +1,497 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Xml;
+using Microsoft.Build.Framework;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GenerateSlnFile : BuildTask
+    {
+        [Required]
+        public string PathToGenerate { get; set; }
+        [Required]
+        public string TemplateFile { get; set; }
+
+        private const string CONFIGURATION_PROPS = "Configurations.props";
+        public override bool Execute()
+        {
+            /** 
+             * Notes about sln generation:
+             * Based on Template Sln File generated by VS2015
+             * Some projects don't have file names matching their sln file
+             * Some projects don't have test projects or src projects
+             * All projects within solution must have a ProjectGUID
+             * Depprojs, PkgProjs shouldn't be added to the sln file
+            **/
+            //expected location of the solution files
+            List<string> slnFileLocations = Directory.EnumerateDirectories(PathToGenerate).ToList();
+            List<string> slnExclusionList = new List<string>()
+            {
+                "src\\Native",
```

Comment:
It is probably best to make this an item group in your targets file so it is a little more extensible.